tracer/src/Datadog.Trace/IAST/Aspects/MongoDB/BsonAspect.cs

```diff
@@ -0,0 +1,150 @@
+// <copyright file="BsonAspect.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Linq;
+using System.Reflection;
+using Datadog.Trace.Iast.Dataflow;
+using Datadog.Trace.Iast.Helpers;
+
+#nullable enable
+
+namespace Datadog.Trace.Iast.Aspects.MongoDB;
+
+#if !NETFRAMEWORK
+/// <summary> BsonAspect class aspect </summary>
+[AspectClass("MongoDB.Bson", AspectType.Sink, VulnerabilityType.NoSqlInjection)]
+[global::System.ComponentModel.Browsable(false)]
+[global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]
+public class BsonAspect
+{
+    /// <summary>
+    /// MongoDB Bson aspect
+    /// </summary>
+    /// <param name="source"> source </param>
+    /// <returns> oui </returns>
+    [AspectMethodReplace("MongoDB.Bson.IO.BsonBinaryReader::ReadString()")]
+    [AspectMethodReplace("MongoDB.Bson.IO.BsonReader::ReadString()")]
+    [AspectMethodReplace("MongoDB.Bson.IO.IBsonReader::ReadString()")]
+    public static object ReadString(object source)
+    {
+        Console.WriteLine(source);
+        return "result";
+    }
+
+    /// <summary>
+    /// MongoDB Deserialize aspect
+    /// </summary>
+    /// <param name="serializer"> the serializer </param>
+    /// <param name="context"> the context </param>
+    /// <returns> the original deserialized object </returns>
+    [AspectMethodReplace("MongoDB.Bson.Serialization.IBsonSerializerExtensions::Deserialize(MongoDB.Bson.Serialization.IBsonSerializer`1<!!0>,MongoDB.Bson.Serialization.BsonDeserializationContext)")]
+    public static object? DeserializeExtension(object serializer, object context)
+    {
+        var result = CallOriginalMethod();
+
+        // The reader can be tainted
+        var reader = context.GetType().GetProperty("Reader")?.GetValue(context);
+        MongoDbHelper.TaintObjectWithJson(result, MongoDbHelper.TaintedLinkedObject(reader));
+
+        return result;
+
+        object? CallOriginalMethod()
+        {
+            try
+            {
+                var typeMethodClass = Type.GetType("MongoDB.Bson.Serialization.IBsonSerializerExtensions, MongoDB.Bson")!;
+                var methodsPublicStatic = typeMethodClass.GetMethods(BindingFlags.Public | BindingFlags.Static);
+                var deserializeMethod = methodsPublicStatic.Where(m => m is { Name: "Deserialize" }).FirstOrDefault();
+                return deserializeMethod?.Invoke(null, new[] { serializer, context });
+            }
+            catch (Exception)
```

Comment:
If the original method call throws an exception we would be modifying the program's behavior here.